signer/src/testing/dummy.rs

```diff
@@ -108,6 +114,112 @@ pub fn block<R: rand::RngCore + ?Sized>(
     bitcoin::Block { header, txdata }
 }
 
+impl Dummy<Faker> for BitcoinTxInfo {
+    fn dummy_with_rng<R: Rng + ?Sized>(config: &Faker, rng: &mut R) -> Self {
+        let output_amount = (500_000..1_000_000_000_u64).choose(rng).unwrap();
+
+        let tx_in = bitcoin::TxIn {
+            previous_output: OutPoint {
+                txid: config.fake_with_rng::<BitcoinTxId, _>(rng).into(),
+                vout: (1..=50u32).choose(rng).unwrap(),
+            },
+            script_sig: ScriptBuf::new(),
+            sequence: bitcoin::Sequence(0),
+            witness: bitcoin::Witness::new(),
+        };
+
+        let tx_out = bitcoin::TxOut {
+            value: Amount::from_sat(output_amount),
+            script_pubkey: config.fake_with_rng::<ScriptPubKey, _>(rng).into(),
+        };
+
+        let tx = bitcoin::Transaction {
+            version: bitcoin::transaction::Version::TWO,
+            lock_time: bitcoin::absolute::LockTime::ZERO,
+            input: vec![tx_in],
+            output: vec![tx_out],
+        };
+
+        tx.fake_with_rng(rng)
+    }
+}
+
+impl Dummy<bitcoin::Transaction> for BitcoinTxInfo {
+    fn dummy_with_rng<R: Rng + ?Sized>(tx: &bitcoin::Transaction, rng: &mut R) -> Self {
+        let fee_rate = (1..=50u64).choose(rng).unwrap();
+
+        let vsize = tx.vsize() as u64;
+        let vin = tx.input.iter().map(|tx_in| tx_in.fake_with_rng(rng));
+
+        BitcoinTxInfo {
+            fee: Some(Amount::from_sat(fee_rate * vsize)),
+            vin: vin.collect(),
+            tx: tx.clone(),
+        }
+    }
+}
+
+impl Dummy<bitcoin::TxIn> for BitcoinTxVin {
+    fn dummy_with_rng<R: Rng + ?Sized>(tx_in: &bitcoin::TxIn, rng: &mut R) -> Self {
+        // Check whether this is a transaction input into a coinbase
+        // transaction.
+        let non_coinbase = !tx_in.previous_output.is_null();
+        let script_pubkey: ScriptPubKey = Faker.fake_with_rng(rng);
+        let output_amount = script_pubkey.minimal_non_dust().to_sat()..Amount::ONE_BTC.to_sat();
+        BitcoinTxVin {
+            txid: Some(tx_in.previous_output.txid),
+            vout: Some(tx_in.previous_output.vout),
+            prevout: non_coinbase.then(|| BitcoinTxVinPrevout {
+                generated: false,
+                value: output_amount.choose(rng).map(Amount::from_sat).unwrap(),
+                script_pubkey: OutputScriptPubKey { script: script_pubkey.into() },
+            }),
+        }
+    }
+}
+
+impl Dummy<Faker> for BitcoinBlockInfo {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        let block_height = (0..i64::MAX as u64).choose(rng).unwrap();
+        BitcoinBlockInfo::random_with_height(block_height, rng)
+    }
+}
+
+impl BitcoinBlockInfo {
+    /// Create a random bitcoin block with the given height
+    pub fn random_with_height<R: Rng + ?Sized>(block_height: u64, rng: &mut R) -> Self {
```

Comment:
Maybe use `BitcoinBlockHeight` instead of u64 ?

Reply:
Yeah, this "new" function replaces an existing one that does basically the same thing as the one that it replaces. The old function used `u64` as the `block_height`, so I basically just copied the type signature. But I'm okay with either changing it or leaving it.

Reply:
I think "use wrapper instead of raw int where possible" can be a good rule of thumb for us